sdks/java/maven-archetypes/examples/build.gradle

```diff
@@ -17,10 +17,44 @@
  */
 
 apply from: project(":").file("build_rules.gradle")
-applyJavaNature()
+applyJavaNature(artifactId: "beam-sdks-java-maven-archetypes-examples")
 
 description = "Apache Beam :: SDKs :: Java :: Maven Archetypes :: Examples"
 
+processResources {
+   filter org.apache.tools.ant.filters.ReplaceTokens, tokens: [
+      'project.version':  version,
+      'bigquery.version': 'v2-rev374-1.22.0',
+      'google-clients.version': '1.22.0',
+      'guava.version': '20.0',
+      'hamcrest.version': '1.3',
+      'jackson.version': '2.8.9',
+      'joda.version': '2.4',
+      'junit.version': '4.12',
+      'maven-compiler-plugin.version': '3.7.0',
+      'maven-exec-plugin.version': '1.6.0',
+      'maven-jar-plugin.version': '3.0.2',
+      'maven-shade-plugin.version': '3.1.0',
+      'pubsub.version': 'v1-rev10-1.22.0',
+      'slf4j.version': '1.7.25',
+      'spark.version': '2.2.1',
+      'hadoop.version': '2.7.3',
+      'maven-surefire-plugin.version': '2.20.1',
+      'mockito.version': '1.9.5'
+   ]
+}
+
+task generateSources(type: Exec) {
+  println "Generating sources for archetype"
+  commandLine './generate-sources.sh'
+}
+
+sourceSets {
+ main {
+  output.dir('src', builtBy: 'generateSources')
+ }
+}
+
 dependencies {
   shadow project(path: ":examples:java", configuration: "shadow")
```

Comment:
This dependency isn't required, hence no dependencies block needed.

Instead we should add a evaluationDependsOn(":examples:java") with a comment stating that we refer to the main and test source sets directly when generating the archetype source.

Reply:
Done. Removed the dependency and added evaluationDependsOn.